### ISSUE TITLE
[3.0.0-beta.32] Duplicating block issues with blocks field type

### ISSUE THREAD

**Issue by joedawson** [payloadcms/payload#6401]
### Describe the Bug

I have a couple of very basic blocks, not doing anything fancy. When I am duplicating any block, whilst it's duplicated, an additional duplicate also appears - however it is empty.

Any blocks that follow the duplicates, also have their state cleared. You can see in the video below, when I duplicate a block the extra duplicate appears as well as the original media block I add is also cleared.

However, not shown in the video, when I publish the value of the blocks re-appear.

[Duplicate block issue](https://github.com/payloadcms/payload/assets/1009696/93345379-d4b7-4cd2-8117-e418adbbbadd)

Here's my fields for reference:
```js
{
  name: 'content',
  label: 'Content',
  type: 'blocks',
  blocks: [
    {
      slug: 'text',
      labels: {
        singular: 'Text',
        plural: 'Text',
      },
      fields: [
        {
          name: 'content',
          label: 'Content',
          type: 'textarea',
          required: true,
        },
      ],
    },
    {
      slug: 'media',
      labels: {
        singular: 'Media',
        plural: 'Media',
      },
      fields: [
        {
          name: 'media',
          label: 'Media',
          type: 'upload',
          relationTo: 'media',
          required: true,
        },
        {
          name: 'caption',
          label: 'Caption',
          type: 'text',
          admin: {
            description: 'Override the media caption (optional)',
          },
        },
      ],
    },
    {
      slug: 'post',
      labels: {
        singular: 'Post',
        plural: 'Posts',
      },
      fields: [
        {
          name: 'post',
          label: 'Post',
          type: 'relationship',
          relationTo: 'posts',
        },
      ],
    },
    {
      slug: 'quote',
      labels: {
        singular: 'Quote',
        plural: 'Quotes',
      },
      fields: [
        {
          name: 'quote',
          label: 'Quote',
          type: 'text',
          required: true,
        },
        {
          name: 'attribution',
          label: 'Attribution',
          type: 'text',
        },
      ],
    },
  ],
}
```

### To Reproduce

- Create a blocks field on a collection
- Duplicate a block

### Payload Version

3.0.0-beta.32

### Adapters and Plugins

db-postgres, storage-s3, plugin-seo

**Comment by JarrodMFlesch** [comment]
@joedawson I am unable to reproduce this issue, I tried with both postgres and mongodb. Could you create a repo that I can clone down to ensure we are seeing the same thing? I would also try recreating this in the simplest collection you possibly can.

Thanks!

**Comment by joedawson** [comment]
Hey @JessChowdhury, thanks for taking a look at this issue. I have created a stripped back version of my project as requested here, which of course contains `package.json` etc: https://github.com/joedawson/block-duplicate-repro

To answer your questions:
- I followed the breaking changes in the `(3.0.0-beta.36)[https://github.com/payloadcms/payload/releases/tag/v3.0.0-beta.36]` release to bump all next, react dependencies etc. along with all payload `3.0.0-beta.36` dependencies.
- I use `pnpm i` to install/reinstall all dependencies
- If I build my application `pnpm run build && pnpm run start` - I **do not** experience the issue, however if I just run the dev server I **do** experience the issue `pnpm run dev`

In short, I experience the duplicate issue I originally reported only when running my dev server.

Let me know if you need any more details - thanks!

**Comment by JessChowdhury** [comment]
@joedawson thank you for the reproduction! I was finally able to recreate this bug by enabling `reactStrictMode: true` in the next.config.

Fix opened in [this PR](https://github.com/payloadcms/payload/pull/6589https://github.com/payloadcms/payload/pull/6589).
